### PR TITLE
all: update for go 1.17

### PR DESCRIPTION
**Pull request by leighmcculloch**

### CODE REVIEW

File: sdk/go.mod [hunk 1]
```diff
@@ -11,3 +11,18 @@ require (
 	github.com/stretchr/testify v1.7.0
 	gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b // indirect
 )
+
+require (
```
Comment (acharb) [inline sdk/go.mod:15]: ❓ are these supposed to be added here?

File: examples/console/go.mod [hunk 2]
```diff
@@ -11,3 +11,22 @@ require (
 	github.com/stellar/go v0.0.0-00010101000000-000000000000
 	gopkg.in/yaml.v2 v2.3.0 // indirect
 )
+
+require (
```
Comment (acharb) [inline examples/console/go.mod:15]: same question here. These didn't show up for me locally when I upgraded to 1.17

Comment (leighmcculloch) [reply examples/console/go.mod:15]: They won't show up until the `go` directive at the top of the file is changed to `1.17`. `go mod tidy` respects the go directive to some degree, so even if you're using go 1.17, if the file says 1.16, it uses the logic from go 1.16 for how to generate the file, but nothing else.

File: sdk/go.mod (hunk 1 above)
Comment (leighmcculloch) [reply sdk/go.mod:15]: Go 1.17 changed the layout of the file a bit. I don't remember the details.

**Review by acharb — APPROVED** [review]
lgtm!

**Pull request merged**